subworkflows/nf-core/abundance_differential_filter/main.nf

```diff
@@ -56,45 +57,80 @@ workflow ABUNDANCE_DIFFERENTIAL_FILTER {
         .combine(ch_contrasts.first()) // Just taking the first contrast
         .multiMap(criteria)
 
-    // Perform normalization and differential analysis
+    // ----------------------------------------------------
+    // Perform differential analysis with limma
+    // ----------------------------------------------------
+
+    LIMMA_NORM(
+        norm_inputs.contrasts.filter{it[0].method == 'limma'}.first(),
+        norm_inputs.samples_and_matrix.filter{it[0].method == 'limma'}
+    )
+
+    LIMMA_DIFFERENTIAL(
+        inputs.contrasts.filter{it[0].method == 'limma'},
+        inputs.samples_and_matrix.filter { it[0].method == 'limma' }
+    )
+
+    // ----------------------------------------------------
+    // Perform differential analysis with deseq2
```

Comment:
```suggestion
    // Run DESeq2 just once to generate a normalised matrix, configure so 
    // that we don't pass subsetting arguments and disregard differential 
    // outputs 
```

Reply:
The normalisations don't get passed to the differential modules, so they should sit separately (which is why they were positioned distinctly before). I'm fine with your arrangement, but if we're mixing them we should probably label them appropriately actually explain why it's done this way.

Reply:
Regarding this, are the normalized matrices in the NORM modules the exact matrices that are used in the DE modules? If so, maybe it would be nice at some point to rearrange the module so NORM -> goes to DE?

Reply:
So, DESeq2 and Limma Voom output normalised matrices (they take raw values), so they don't need to be passed normalised matrices.

So, originally on differentialabundance we just took the first output of DESEQ2_DIFFERENTIAL and used the normalised matrix from there. But it caused problems, since that first contrast might have options (the exclude samples ones) that lead to the matrix being subsetted, so the normalised matrix we got did not always contain all samples.

For that reason we developed this workaround, where the differential module is run via a separate alias, just once, not passing any subsetting options (see the config here: https://github.com/nf-core/differentialabundance/blob/3dd360fed0dca1780db1bdf5dce85e5258fa2253/conf/modules.config#L162), and we take the normalised output from there, disregarding the differential.

Reply:
Why for DIFF you do have cases where you remove samples, whereas for NORM you want the full set? Is it because you need them for GSEA? Or that the plotting need the entire set?